### ISSUE TITLE
1.2.2 DualPivotQuickSort optimization is buggy

### ISSUE THREAD

**Issue by jeffrey-easyesi** [mintern-java/primitive#6]
The JDK9 "optimization" of DualPivotQuickSort, which was merged into primitives-1.2.2, is buggy. If there is a sequence of 2+ identical elements at the end of the array, those elements may get replaced with 0. Example:
```java
int[] arr = {
    999,267,269,356,443,999,100,101,102,103,104,105,106,107,108,109,
    110,111,999,114,115,116,117,118,119,120,121,122,123,124,125,126,
    127,128,130,131,132,129,133,134,135,136,137,138,139,140,141,142,
    143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,
    159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,
    175,176,178,180,181,182,183,184,999,218,221,222,224,225,999,112,
    113,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,
    200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,999,
    227,229,231,233,235,237,244,245,246,249,263,265,999,223,999,228,
    234,266,999,357,359,360,361,362,358,363,364,365,366,367,368,369,
    370,371,372,373,374,375,376,377,378,379,380,381,382,383,384,385,
    386,387,388,389,390,391,392,393,394,396,395,397,398,399,400,401,
    402,403,404,405,406,407,408,409,410,411,412,413,414,415,416,417,
    418,419,420,421,422,423,424,425,426,427,428,429,430,431,432,433,
    434,436,435,437,438,439,440,441,442,999,217,230,232,264,999,268,
    999,219,220,250,251,252,253,254,255,256,257,258,259,260,261,262,
    999,179,999,236,999,216,999,226,238,239,240,241,242,999,270,271,
    272,273,274,275,276,277,278,279,280,281,282,283,284,285,286,287,
    288,289,291,290,292,293,294,295,296,297,298,299,300,301,303,302,
    304,305,306,307,308,309,310,311,312,313,314,316,315,318,317,319,
    320,321,322,323,324,325,326,327,328,329,330,331,332,333,334,336,
    335,337,338,339,340,341,342,343,344,345,346,347,348,349,350,351,
    352,353,354,355,177,999,243,999,444,445,446,447,448,449,450,452,
    453,454,451,455,456,457,458,459,460,461,462,463,464,465,466,467,
    468,469,470,471,472,473,474,475,476,477,478,479,480,481,482,483,
    484,485,486,487,488,489,490,491,492,493,495,494,496,497,498,499,
    500,501,502,999,215,999,248,247,999,999,999,999,999,999,999,999
};
Primitive.sort(arr, Integer::compare, false);
System.out.println(Arrays.stream(arr).mapToObj(Integer::toString).collect(Collectors.joining(",")));
```

This prints `100,101,102,103, ... 999,999,999,999,0,0,0,0,0,0,0,0` despite there being no 0s in the original array. IMO, it would be best to revert to the known good JDK8 algorithm for now.

**Comment by jeffrey-easyesi** [comment]
This has already been reported to OpenJDK: https://bugs.openjdk.java.net/browse/JDK-8154049

**Comment by ajshankar** [comment]
Nice work Jeff!

**Comment by mintern** [comment]
Thanks for the excellent bug report and for the test! I've just added a commit that reproduces the bug using that test: 08e896c. I'll be releasing a fix shortly.

Is that test code your own? If so, will you license it under the [GPLv2 with Classpath Exception](http://openjdk.java.net/legal/gplv2+ce.html) so that I can include it with the rest of the tests?

**Comment by jeffrey-easyesi** [comment]
I created the array at work, derived from data on our server (but munged to the extent that no meaningful information should remain), so I presume it belongs to the company.

**Comment by ajshankar** [comment]
I don't know what it takes to release the data (other than me saying it), but I'm certainly fine with us re-licensing these numbers.

**Comment by mintern** [comment]
Thank you both! Everlaw will retain copyright and "All Rights Reserved", but it will be licensed under the GPLv2+CE like the rest of the code.

**Comment by mintern** [comment]
Fixed by #7. I published the 1.2.3 artifact to Sonatype, so it should land in the Maven Central repo soon.

Thanks again!
